packages/components/src/components/dropdown/dropdown.tsx

```diff
@@ -1,18 +1,72 @@
-import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import ReactDOM from 'react-dom';
+import classNames from 'classnames';
 import { CSSTransition } from 'react-transition-group';
 import { mobileOSDetect, getPosition } from '@deriv/shared';
-import { listPropType, findNextFocusableNode, findPreviousFocusableNode } from './dropdown';
-import Items from './items.jsx';
-import DisplayText from './display-text.jsx';
+import { TList, findNextFocusableNode, findPreviousFocusableNode, TListItem } from './utility';
+import Items from './items';
+import DisplayText from './display-text';
 import Text from '../text/text';
-import { useBlockScroll, useOnClickOutside } from '../../hooks';
+import { IClickEvent, useBlockScroll, useOnClickOutside } from '../../hooks';
 import ThemedScrollbars from '../themed-scrollbars/themed-scrollbars';
 import Icon from '../icon/icon';
 
-const DropdownList = React.forwardRef((props, list_ref) => {
+type TDropdown = {
+    className?: string;
+    classNameDisplay?: string;
+    classNameHint?: string;
+    classNameItems?: string;
+    classNameLabel?: string;
+    disabled?: boolean;
+    error?: string;
+    handleBlur?: {
+        (e: React.FocusEvent<any>): void;
+        <T = string | any>(fieldOrEvent: T): T extends string ? (e: any) => void : void;
```

Comment:
```suggestion
    handleBlur?: {
        (e: React.FocusEvent<HTMLElement>): void;
        <T = string | any>(fieldOrEvent: T): T extends string ? (e: React.FocusEvent<HTMLElement>) => void : void;
```
@niloofar-deriv could you please try this?

Reply:
sure will change this :pray: